From 30d980c6f1e7b3e573a3d01d891648ab7496e585 Mon Sep 17 00:00:00 2001
From: Tom Marshall <tdm@cyngn.com>
Date: Fri, 12 Dec 2014 16:35:56 -0800
Subject: [PATCH 2/6] sepolicy: New type sdcard_posix for labeled filesystems

 * Need this because sdcard_external is an alias for vfat

Change-Id: I804ebb0fcf643d603b1a02ee7a54e6d5b6b46294
---
 public/domain.te | 2 +-
 public/file.te   | 1 +
 public/vold.te   | 1 +
 3 files changed, 3 insertions(+), 1 deletion(-)

diff --git a/public/domain.te b/public/domain.te
index d2b370a..6151ebd 100644
--- a/public/domain.te
+++ b/public/domain.te
@@ -413,7 +413,7 @@ neverallow * rootfs:file { create write setattr relabelto append unlink link ren
 
 # Restrict context mounts to specific types marked with
 # the contextmount_type attribute.
-neverallow * {fs_type -contextmount_type}:filesystem relabelto;
+neverallow * {fs_type -contextmount_type -sdcard_posix}:filesystem relabelto;
 
 # Ensure that context mount types are not writable, to ensure that
 # the write to /system restriction above is not bypassed via context=
diff --git a/public/file.te b/public/file.te
index 057af41..993b310 100644
--- a/public/file.te
+++ b/public/file.te
@@ -61,6 +61,7 @@ type shm, fs_type;
 type mqueue, fs_type;
 type fuse, sdcard_type, fs_type, mlstrustedobject;
 type sdcardfs, sdcard_type, fs_type, mlstrustedobject;
+type sdcard_posix, sdcard_type, fs_type, mlstrustedobject;
 type vfat, sdcard_type, fs_type, mlstrustedobject;
 type debugfs, fs_type;
 type debugfs_mmc, fs_type, debugfs_type;
diff --git a/public/vold.te b/public/vold.te
index 81ee28c..4718051 100644
--- a/public/vold.te
+++ b/public/vold.te
@@ -38,6 +38,7 @@ allow vold devpts:chr_file rw_file_perms;
 allow vold rootfs:dir mounton;
 allow vold sdcard_type:dir mounton; # TODO: deprecated in M
 allow vold sdcard_type:filesystem { mount remount unmount }; # TODO: deprecated in M
+allow vold sdcard_posix:filesystem { relabelto relabelfrom };
 allow vold sdcard_type:dir create_dir_perms; # TODO: deprecated in M
 allow vold sdcard_type:file create_file_perms; # TODO: deprecated in M
 
-- 
2.1.2

